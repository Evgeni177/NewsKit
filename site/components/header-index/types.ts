import {ImageProps} from 'newskit';

export interface HeaderIndexProps {
  title: string;
  children: React.ReactNode;
  media: ImageProps;
}
