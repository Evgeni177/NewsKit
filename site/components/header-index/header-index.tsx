import {Cell, Grid, TextBlock, Image, styled, Block, getColorFromTheme} from 'newskit';
import React from 'react';
import {HeaderIndexProps} from './types';

export const HeaderIndex: React.FC<HeaderIndexProps> = ({
  title,
  children,
  media,
}) => {

  const StyledGrid = styled(Grid)`
    align-items: center;
    background-color: ${getColorFromTheme('interface020')};
  `;

  return (
    <StyledGrid xsRowGutter="space020" mdRowGutter="space030">
      <Cell lgOrder={2} xs={12} md={6} lg={5} mdOffset={6} lgOffset={0}>
        <Image src={media.src} alt=""/>
      </Cell>
      <Cell xs={12} md={8} lg={6} mdOffset={1}>
        <Block spaceStack={{xs: 'space060', md: 'space070'}}>
          <TextBlock
            as="h1"
            stylePreset="inkContrast"
            typographyPreset={{
              xs: 'editorialHeadline060',
              md: 'editorialHeadline080',
            }}
          >
            {title}
          </TextBlock>
        </Block>
        <TextBlock
          stylePreset="inkBase"
          typographyPreset={{
            xs: 'editorialSubheadline010',
            md: 'editorialSubheadline020',
          }}
        >
          {children}
        </TextBlock>
      </Cell>
    </StyledGrid>
  );
};
