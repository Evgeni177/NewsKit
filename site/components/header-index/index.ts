export {HeaderIndex} from './header-index';
