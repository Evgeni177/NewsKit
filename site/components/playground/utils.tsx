import React from 'react';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const isComponent = (value: any): value is React.ComponentType =>
  Boolean(value && (value.$$typeof || value.displayName));

export const withProps = (
  Component: React.ComponentType,
  props: Record<string, unknown>,
  displayName?: string,
) => {
  const Comp: React.FC = p => <Component {...props} {...p} />;
  Comp.displayName = displayName || Component.displayName;
  return Comp;
};
