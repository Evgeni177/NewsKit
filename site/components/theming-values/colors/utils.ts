import {Theme, ThemeProp} from 'newskit';
import {ThemeColor} from './types';
import {getContrast, getBrightness, getContrastRating} from './color-utils';

export const getBorderRadius = ({
  first,
  last,
  theme,
}: {
  first?: boolean;
  last?: boolean;
  theme?: Theme;
}) => {
  const fn = ({
    theme: {
      borderRadius: {borderRadiusRounded020: radius},
    },
  }: ThemeProp) =>
    [
      first ? radius : 0,
      first ? radius : 0,
      last ? radius : 0,
      last ? radius : 0,
    ].join(' ');

  if (theme) {
    return fn({theme});
  }

  return fn;
};

export const getTextColor = ({
  theme,
  value,
}: // brightness,
{theme: Theme} & Pick<ThemeColor, 'brightness' | 'value'>) =>
  // brightness < 0.4 ? theme.colors.inkInverse : theme.colors.inkContrast;
  getContrast(value, theme.colors.inkInverse) <
  getContrast(value, theme.colors.inkContrast)
    ? theme.colors.inkInverse
    : theme.colors.inkContrast;

export const getColorGroup = (name: string) => (/[a-z]+/.exec(name) || [])[0];

export const mapColorObjects = (
  theme: Theme,
  paletteColorMap?: Record<string, ThemeColor[]>,
) => ([name, value]: [string, string]) => {
  const brightness = getBrightness(value);
  const contrastRatio = getContrast(value, theme.colors.interface010);
  const contrastRating = getContrastRating(contrastRatio);

  return {
    name,
    value,
    group: getColorGroup(name),
    brightness,
    contrastRating,
    contrastRatio,
    parentColor:
      paletteColorMap && paletteColorMap[value]
        ? paletteColorMap[value][0]
        : undefined,
  } as ThemeColor;
};

export const groupColorObjects = (
  acc: Record<string, ThemeColor[]>,
  color: ThemeColor,
) => {
  // eslint-disable-next-line no-multi-assign
  const group = (acc[color.group] = acc[color.group] || []);
  group.push(color);
  return acc;
};

export const createPaletteColorMap = (paletteColors: ThemeColor[]) =>
  paletteColors.reduce(
    (acc, color) => {
      // eslint-disable-next-line no-multi-assign
      const accGroup = (acc[color.value] = acc[color.value] || []);
      accGroup.push(color);
      return acc;
    },
    {} as Record<string, ThemeColor[]>,
  );
