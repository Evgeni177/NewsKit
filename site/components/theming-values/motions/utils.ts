export const CANVAS_SIZE = 56;

const parseCubicBezier = (value: string) =>
  value
    .replace('cubic-bezier(', '')
    .replace(')', '')
    .split(',')
    .map(v => Number.parseFloat(v.trim()));

// NEXT 2 FUNCTIONS ARE COPIED FROM http://dbaron.org/css/timing-function-graphs
export function bezier(x1: number, y1: number, x2: number, y2: number) {
  // Cubic bezier with control points (0, 0), (x1, y1), (x2, y2), and (1, 1).
  function xForT(t: number) {
    const omt = 1 - t;
    return 3 * omt * omt * t * x1 + 3 * omt * t * t * x2 + t * t * t;
  }
  function yForT(t: number) {
    const omt = 1 - t;
    return 3 * omt * omt * t * y1 + 3 * omt * t * t * y2 + t * t * t;
  }
  function tForX(x: number) {
    // Binary subdivision.
    let mint = 0;
    let maxt = 1;
    for (let i = 0; i < 30; ++i) {
      const guesst = (mint + maxt) / 2;
      const guessx = xForT(guesst);
      if (x < guessx) maxt = guesst;
      else mint = guesst;
    }
    return (mint + maxt) / 2;
  }
  return function bezierClosure(x: number) {
    if (x === 0) return 0;
    if (x === 1) return 1;
    return yForT(tForX(x));
  };
}

export const updateCanvas = (
  ctx: CanvasRenderingContext2D,
  timingFn: string,
  color: string,
) => {
  let x1 = 0;
  let y1 = 0;
  let x2 = 0;
  let y2 = 0;
  switch (timingFn) {
    case 'ease':
      x1 = 0.25;
      y1 = 0.1;
      x2 = 0.25;
      y2 = 1.0;
      break;
    case 'linear':
      x1 = 0.0;
      y1 = 0.0;
      x2 = 1.0;
      y2 = 1.0;
      break;
    case 'ease-in':
      x1 = 0.42;
      y1 = 0.0;
      x2 = 1.0;
      y2 = 1.0;
      break;
    case 'ease-out':
      x1 = 0.0;
      y1 = 0.0;
      x2 = 0.58;
      y2 = 1.0;
      break;
    case 'ease-in-out':
      x1 = 0.42;
      y1 = 0.0;
      x2 = 0.58;
      y2 = 1.0;
      break;
    default:
      [x1, y1, x2, y2] = parseCubicBezier(timingFn);
      break;
  }

  const tf = bezier(x1, y1, x2, y2);

  const gCurrentTFPoints = [];

  for (let x = 0; x <= CANVAS_SIZE; ++x) {
    gCurrentTFPoints[x] = tf(x / CANVAS_SIZE) * CANVAS_SIZE;
  }

  ctx.clearRect(0, 0, CANVAS_SIZE, CANVAS_SIZE);

  // Draw the graph
  ctx.strokeStyle = color;
  ctx.lineWidth = 2;
  ctx.beginPath();
  ctx.moveTo(0, CANVAS_SIZE - gCurrentTFPoints[0]);
  for (let x = 1; x <= CANVAS_SIZE; ++x) {
    ctx.lineTo(x, CANVAS_SIZE - gCurrentTFPoints[x]);
  }
  ctx.stroke();
};

export const percentFrom = (total: number, value: number) =>
  Math.floor(value / (total / 100));

export const getTotalDuration = (stayDuration: number, maxDuration: number) =>
  2 * stayDuration + 2 * maxDuration;
