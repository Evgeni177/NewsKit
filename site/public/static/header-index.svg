
<svg width="403" height="347" viewBox="0 0 403 347" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M142.705 208.406L27.1689 62.4072" stroke="#2B4AAB" stroke-width="4" stroke-dasharray="4 8"/>
<path d="M207.567 101.251L321.076 82.499" stroke="#2B4AAB" stroke-width="4" stroke-dasharray="4 8"/>
<path d="M142.705 208.406L321.076 83.1685" stroke="#2B4AAB" stroke-width="4"/>
<path d="M207.567 101.921L142.705 208.406" stroke="#2B4AAB" stroke-width="4"/>
<path d="M26.4941 62.4072L206.892 101.921" stroke="#2B4AAB" stroke-width="4"/>
<path d="M26.8315 88.526C12.0921 88.526 0.143433 76.6822 0.143433 62.0721C0.143433 47.462 12.0921 35.6182 26.8315 35.6182C41.5709 35.6182 53.5195 47.462 53.5195 62.0721C53.5195 76.6822 41.5709 88.526 26.8315 88.526Z" fill="#3358CC"/>
<path d="M208.243 120.003C198.168 120.003 190 111.757 190 101.586C190 91.4142 198.168 83.1685 208.243 83.1685C218.318 83.1685 226.485 91.4142 226.485 101.586C226.485 111.757 218.318 120.003 208.243 120.003Z" fill="#213A82"/>
<path d="M142.705 249.259C119.943 249.259 101.491 230.818 101.491 208.071C101.491 185.324 119.943 166.883 142.705 166.883C165.468 166.883 183.92 185.324 183.92 208.071C183.92 230.818 165.468 249.259 142.705 249.259Z" fill="#87A4FC"/>
<mask id="mask0" mask-type="alpha" maskUnits="userSpaceOnUse" x="260" y="22" width="121" height="121">
<path d="M290.482 30.7573C261.722 47.1943 251.905 83.6605 268.554 112.207C285.203 140.753 322.015 150.57 350.775 134.133C379.535 117.696 389.353 81.2301 372.703 52.6836C356.054 24.1371 319.242 14.3204 290.482 30.7573Z" fill="white"/>
</mask>
<g mask="url(#mask0)">
<path d="M290.493 30.7683C261.727 47.2114 251.9 83.6777 268.543 112.218C285.186 140.758 321.998 150.565 350.764 134.122C379.53 117.679 389.358 81.2126 372.714 52.6723C356.071 24.132 319.26 14.3252 290.493 30.7683Z" fill="#2B4AAB"/>
</g>
<mask id="mask1" mask-type="alpha" maskUnits="userSpaceOnUse" x="260" y="22" width="121" height="121">
<path d="M290.482 30.7573C261.722 47.1943 251.905 83.6605 268.554 112.207C285.203 140.753 322.015 150.57 350.775 134.133C379.535 117.696 389.353 81.2301 372.703 52.6836C356.054 24.1371 319.242 14.3204 290.482 30.7573Z" fill="white"/>
</mask>
<g mask="url(#mask1)">
<path d="M290.493 30.7683C261.727 47.2114 251.9 83.6777 268.543 112.218C285.186 140.758 321.998 150.565 350.764 134.122C379.53 117.679 389.358 81.2126 372.714 52.6723C356.071 24.132 319.26 14.3252 290.493 30.7683Z" fill="#2B4AAB"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M377.623 58.6813C364.103 29.7391 345.196 27.194 344.748 27.0893C331.658 24.031 303.611 19.3806 287.714 30.3083C271.818 41.2361 264.344 83.2736 287.711 96.4219C308.828 108.303 310.938 127.359 324.504 130.879C338.071 134.399 354.97 130.484 368.834 118.627C382.697 106.769 391.143 87.6235 377.623 58.6813Z" fill="#3358CC"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M368.059 31.5785C386.583 46.8576 384.734 60.9312 384.722 61.2733C384.391 71.2664 384.059 85.056 372.942 94.3767C361.825 103.697 339.612 103.312 339.137 103.272C329.736 102.466 325.918 88.2584 318.253 80.6625C310.017 72.5015 303.357 65.3423 303.445 54.9181C303.532 44.494 314.466 34.3008 322.471 26.368C330.276 18.6334 349.536 16.2995 368.059 31.5785Z" fill="#577FFB"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M366.085 39.5354C374.57 46.5342 373.724 52.9809 373.718 53.1376C373.566 57.7151 373.414 64.0317 368.322 68.3012C363.23 72.5708 353.055 72.3944 352.837 72.3757C348.531 72.0069 346.782 65.4986 343.271 62.0192C339.498 58.2809 336.447 55.0015 336.488 50.2265C336.528 45.4515 341.536 40.7824 345.203 37.1486C348.778 33.6056 357.6 32.5365 366.085 39.5354Z" fill="#87A4FC"/>
</g>
<path d="M143.936 326.853C154.32 327.213 163.684 328.251 170.426 329.635C173.801 330.328 176.495 331.103 178.326 331.912C179.245 332.317 179.917 332.719 180.349 333.102C180.786 333.491 180.91 333.798 180.902 334.024C180.894 334.25 180.749 334.549 180.286 334.906C179.828 335.259 179.128 335.613 178.184 335.954C176.3 336.634 173.558 337.22 170.142 337.678C163.32 338.593 153.906 338.981 143.522 338.622C133.138 338.262 123.774 337.224 117.032 335.84C113.657 335.147 110.963 334.372 109.132 333.564C108.213 333.158 107.541 332.757 107.109 332.373C106.672 331.985 106.548 331.677 106.556 331.451C106.564 331.225 106.709 330.926 107.172 330.569C107.63 330.216 108.33 329.862 109.274 329.521C111.158 328.841 113.9 328.255 117.316 327.797C124.138 326.882 133.552 326.494 143.936 326.853ZM320.142 312.864C332.561 313.294 343.775 314.294 351.862 315.563C355.909 316.197 359.154 316.896 361.372 317.618C362.485 317.98 363.31 318.338 363.844 318.681C364.416 319.048 364.494 319.291 364.491 319.377C364.488 319.464 364.393 319.701 363.797 320.027C363.24 320.332 362.391 320.633 361.256 320.917C358.993 321.483 355.706 321.956 351.624 322.309C343.467 323.015 332.211 323.238 319.792 322.808C307.373 322.379 296.159 321.378 288.072 320.11C284.025 319.475 280.78 318.776 278.562 318.055C277.449 317.693 276.624 317.334 276.09 316.991C275.518 316.624 275.44 316.382 275.443 316.295C275.446 316.208 275.541 315.971 276.137 315.645C276.694 315.34 277.543 315.039 278.678 314.755C280.941 314.189 284.228 313.716 288.31 313.363C296.467 312.657 307.723 312.434 320.142 312.864ZM27.0634 301.806C33.3934 302.025 39.1047 302.529 43.2192 303.164C45.2798 303.482 46.9216 303.83 48.0356 304.186C48.5965 304.365 48.995 304.538 49.2418 304.693C49.3263 304.746 49.3792 304.788 49.4121 304.819C49.3772 304.846 49.3214 304.885 49.2334 304.932C48.9762 305.07 48.5666 305.214 47.9945 305.354C46.8581 305.632 45.1959 305.866 43.1178 306.041C38.9686 306.39 33.2358 306.498 26.9058 306.279C20.5759 306.059 14.8646 305.556 10.75 304.92C8.68939 304.602 7.04766 304.254 5.9336 303.898C5.37276 303.719 4.97426 303.547 4.7274 303.391C4.64298 303.338 4.59 303.296 4.55718 303.266C4.59206 303.238 4.64787 303.2 4.73583 303.152C4.99302 303.014 5.40267 302.87 5.97475 302.73C7.11113 302.452 8.77338 302.218 10.8514 302.044C15.0006 301.694 20.7335 301.587 27.0634 301.806ZM230.352 288.254C230.352 288.277 230.333 288.391 230.074 288.579C229.825 288.761 229.428 288.954 228.871 289.143C227.763 289.521 226.137 289.849 224.096 290.106C220.022 290.619 214.387 290.837 208.149 290.637L207.77 290.624C207.77 290.624 207.77 290.624 207.77 290.624C201.445 290.405 195.744 289.8 191.643 289.002C189.589 288.603 187.959 288.159 186.857 287.701C186.304 287.471 185.913 287.25 185.671 287.048C185.42 286.839 185.409 286.721 185.41 286.698C185.411 286.675 185.429 286.561 185.688 286.372C185.937 286.19 186.334 285.998 186.891 285.808C187.999 285.431 189.625 285.103 191.666 284.846C195.74 284.333 201.375 284.115 207.613 284.315C207.613 284.315 207.613 284.315 207.613 284.315L207.992 284.327C207.992 284.327 207.992 284.327 207.992 284.327C214.318 284.546 220.018 285.152 224.119 285.949C226.174 286.349 227.804 286.793 228.905 287.25C229.458 287.48 229.849 287.702 230.091 287.904C230.342 288.113 230.353 288.23 230.352 288.254Z" fill="#BBC3C9" stroke="#BBC3C9"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M207.232 286.54L26.525 304.026L207.232 286.54ZM140.94 333.553L319.967 317.835L140.94 333.553Z" fill="#BBC3C9"/>
<path d="M140.94 333.553L319.967 317.835M207.232 286.54L26.525 304.026L207.232 286.54Z" stroke="#BBC3C9" stroke-width="2"/>
<path d="M140.942 333.552L207.912 286.565" stroke="#BBC3C9" stroke-width="2"/>
<path d="M207.88 287.477L319.967 317.842" stroke="#BBC3C9" stroke-width="2" stroke-dasharray="4 7"/>
<path d="M25.1152 304.883L143.731 332.735" stroke="#BBC3C9" stroke-width="2" stroke-dasharray="4 8"/>
</svg>
