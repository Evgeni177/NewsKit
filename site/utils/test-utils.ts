import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme = renderToFragmentWithThemeFactory(
  docsThemeLight,
);
