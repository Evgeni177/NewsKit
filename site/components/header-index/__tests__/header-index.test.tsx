import {renderToFragmentWithTheme} from '../../../utils/test-utils';
import {HeaderIndex} from '../header-index';

describe('HeaderIndex', () => {
  test('renders without crashing', () => {
    const props = {
      title: 'Foundations',
      media: {
        src: 'test',
      },
      children: 'description',
    };

    const fragment = renderToFragmentWithTheme(HeaderIndex, props);
    expect(fragment).toMatchSnapshot();
  });
});
