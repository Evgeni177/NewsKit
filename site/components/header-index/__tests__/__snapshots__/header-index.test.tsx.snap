// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderIndex renders without crashing 1`] = `
<DocumentFragment>
  .emotion-0 {
  background-color: #F4F4F4;
}

@media screen and (max-width: 1023px) {
  .emotion-0 {
    margin-bottom: 48px;
  }
}

@media screen and (min-width: 1024px) {
  .emotion-0 {
    margin-bottom: 64px;
  }
}

.emotion-1 {
  width: 100%;
  max-width: 1920px;
  margin: 0 auto;
}

.emotion-2 {
  box-sizing: border-box;
  -webkit-background-clip: padding-box;
  background-clip: padding-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media screen {
  .emotion-2 {
    margin: -8px 4px 0 4px;
  }
}

@media screen and (min-width: 480px) {
  .emotion-2 {
    margin: -8px 4px 0 4px;
  }
}

@media screen and (min-width: 768px) {
  .emotion-2 {
    margin: -12px 4px 0 4px;
  }
}

@media screen and (min-width: 1024px) {
  .emotion-2 {
    margin: -12px 12px 0 12px;
  }
}

@media screen and (min-width: 1440px) {
  .emotion-2 {
    margin: -12px 12px 0 12px;
  }
}

@media screen and (min-width: 1024px) {
  .emotion-2 {
    min-height: 440px;
  }
}

.emotion-3 {
  box-sizing: border-box;
  -webkit-background-clip: padding-box;
  background-clip: padding-box;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

@media screen {
  .emotion-3 {
    padding: 0 12px;
    margin-top: 8px;
    margin-left: 8.333333333333334%;
    -webkit-flex-basis: 83.33333333333334%;
    -ms-flex-preferred-size: 83.33333333333334%;
    flex-basis: 83.33333333333334%;
    max-width: 83.33333333333334%;
  }
}

@media screen and (min-width: 480px) {
  .emotion-3 {
    margin-left: 33.333333333333336%;
    -webkit-flex-basis: 66.66666666666667%;
    -ms-flex-preferred-size: 66.66666666666667%;
    flex-basis: 66.66666666666667%;
    max-width: 66.66666666666667%;
  }
}

@media screen and (min-width: 768px) {
  .emotion-3 {
    margin-top: 12px;
    margin-left: 50%;
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    max-width: 50%;
  }
}

@media screen and (min-width: 1024px) {
  .emotion-3 {
    -webkit-order: 2;
    -ms-flex-order: 2;
    order: 2;
    margin-left: 0;
    -webkit-flex-basis: 41.66666666666667%;
    -ms-flex-preferred-size: 41.66666666666667%;
    flex-basis: 41.66666666666667%;
    max-width: 41.66666666666667%;
  }
}

.emotion-4 {
  width: 100%;
}

.emotion-5 {
  border-radius: 8px;
  position: relative;
  width: 100%;
  padding-top: 0;
  height: 0;
  border-radius: 8px;
  background-color: #F4F4F4;
}

.emotion-5 svg {
  fill: #BBC3C9;
}

.emotion-6 {
  top: 0;
  left: 0;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  margin: 0;
}

.emotion-7 {
  display: none;
  height: auto;
  width: 100%;
  -webkit-animation: fadeIn 300ms;
  animation: fadeIn 300ms;
  border-radius: inherit;
}

.emotion-8 {
  box-sizing: border-box;
  -webkit-background-clip: padding-box;
  background-clip: padding-box;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

@media screen {
  .emotion-8 {
    padding: 0 12px;
    margin-top: 8px;
    -webkit-flex-basis: 91.66666666666667%;
    -ms-flex-preferred-size: 91.66666666666667%;
    flex-basis: 91.66666666666667%;
    max-width: 91.66666666666667%;
  }
}

@media screen and (min-width: 768px) {
  .emotion-8 {
    margin-top: 12px;
    margin-left: 8.333333333333334%;
    -webkit-flex-basis: 66.66666666666667%;
    -ms-flex-preferred-size: 66.66666666666667%;
    flex-basis: 66.66666666666667%;
    max-width: 66.66666666666667%;
  }
}

@media screen and (min-width: 1024px) {
  .emotion-8 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    max-width: 50%;
  }
}

@media screen and (max-width: 767px) {
  .emotion-9 {
    margin-bottom: 32px;
  }
}

@media screen and (min-width: 768px) {
  .emotion-9 {
    margin-bottom: 40px;
  }
}

.emotion-10 {
  margin: 0;
  padding: 1px 0;
  color: #162740;
}

.emotion-10 svg {
  fill: #162740;
}

@media screen and (max-width: 767px) {
  .emotion-10 {
    font-family: 'Poppins',sans-serif;
    font-size: 40px;
    line-height: 1.2;
    font-weight: 700;
    letter-spacing: 0;
  }

  .emotion-10::before {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-bottom: -0.140625em;
  }

  .emotion-10::after {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-top: -0.390625em;
  }
}

@media screen and (min-width: 768px) {
  .emotion-10 {
    font-family: 'Poppins',sans-serif;
    font-size: 60px;
    line-height: 1.2;
    font-weight: 700;
    letter-spacing: 0;
  }

  .emotion-10::before {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-bottom: -0.140625em;
  }

  .emotion-10::after {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-top: -0.390625em;
  }
}

@media screen and (max-width: 1023px) {
  .emotion-11 {
    margin-bottom: 48px;
  }
}

@media screen and (min-width: 1024px) {
  .emotion-11 {
    margin-bottom: 0;
  }
}

.emotion-12 {
  margin: 0;
  padding: 1px 0;
  color: #555E74;
  font-family: 'Poppins',sans-serif;
  font-size: 20px;
  line-height: 1.6;
  font-weight: 500;
  letter-spacing: 0;
}

.emotion-12 svg {
  fill: #555E74;
}

.emotion-12::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.34062500000000007em;
}

.emotion-12::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.5906250000000001em;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-3"
        >
          <div
            class="emotion-4"
          >
            <div
              class="emotion-5"
            >
              <div
                class="emotion-6"
              />
              <img
                alt=""
                class="emotion-7"
                src="test"
              />
            </div>
          </div>
        </div>
        <div
          class="emotion-8"
        >
          <div
            class="emotion-9"
          >
            <h1
              class="emotion-10"
            >
              Foundations
            </h1>
          </div>
          <div
            class="emotion-11"
          >
            <p
              class="emotion-12"
            >
              description
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
